### DISCUSSION TITLE
How to upgrade vite-plugin

### DISCUSSION THREAD

**Discussion by fprijate** [quasarframework/quasar#15135] [General - Components / Directives / etc] [answered]
New project created with **"npm init quasar"**  has @quasar/vite-plugin  1.3.0 , which supports  Vite 4.
How do I upgrade vite-plugin on old project. **Quasar upgrade** doesn't work.

**Comment by rstoenescu** [comment, marked as answer]
Hi,

You are mixing things here.

`yarn create quasar` / `npm init quasar` uses Quasar CLI (with Webpack or Vite depending on your choices). It creates a Quasar CLI project, not a plain Vite project. Only plain Vite projects use our vite-plugin directly. The Quasar CLI way (which is what we highly recommend) ships with our vite-plugin too, but as one of its dependencies rather than a direct dep of your project.

The vite-plugin now has support for Vite 2, 3 & 4, but the Quasar CLI is still using Vite 2 -- and work under way for the next major of the CLI with support for Vite 4, but there are other features that we want to ship when we make the jump (so it'll take a bit more time). One example is SSR with ESM (instead of CommonJs), which is no easy task (since it requires some changes in both the UI and CLI packages). The main priority is to make the transition as smooth as possible, with no headaches. There is a ticket opened [here](https://github.com/quasarframework/quasar/issues/14077) for Vite 4 support in Quasar CLI if you want to check it out. It also contains a temporary workaround for using Vite 3/4 with current Quasar CLI.
